framework/src/positions/ParsedSelectionPositions.C

```diff
@@ -0,0 +1,168 @@
+  return params;
+}
+
+ParsedSelectionPositions::ParsedSelectionPositions(const InputParameters & parameters)
+  : Positions(parameters),
+    NonADFunctorInterface(this),
+    BlockRestrictable(this),
+    FunctionParserUtils(parameters),
+    _expression(getParam<std::string>("expression")),
+    _xyzt({"x", "y", "z", "t"}),
+    _functor_names(getParam<std::vector<MooseFunctorName>>("functor_names")),
+    _n_functors(_functor_names.size()),
+    _functor_symbols(getParam<std::vector<std::string>>("functor_symbols"))
+{
+  if (getParam<ExecFlagEnum>("execute_on").contains(EXEC_NONE))
+    paramError("execute_on",
+               "NONE execution flag not supported. Most functors (functions, variables, spatial "
+               "user objects for example) are not initialized at construction.");
+
+  // sanity checks
+  if (!_functor_symbols.empty() && (_functor_symbols.size() != _n_functors))
+    paramError("functor_symbols", "functor_symbols must be the same length as functor_names.");
+
+  // Make sure functors are not x, y, z, or t
+  for (const auto & name : _functor_symbols)
+    if (std::find(_xyzt.begin(), _xyzt.end(), name) != _xyzt.end())
+      paramError("functor_symbols", "x, y, z, and t cannot be used in 'functor_symbols'.");
+  for (const auto & name : _functor_names)
+    if (std::find(_xyzt.begin(), _xyzt.end(), name) != _xyzt.end())
+      paramError(
+          "functor_names",
+          "x, y, z, and t cannot be used in 'functor_names'. Use functor_symbols to disambiguate");
+
+  // build variables argument
+  std::string variables;
+
+  // adding functors to the expression
+  if (_functor_symbols.size())
+    for (const auto & symbol : _functor_symbols)
+      variables += (variables.empty() ? "" : ",") + symbol;
+  else
+    for (const auto & name : _functor_names)
+      variables += (variables.empty() ? "" : ",") + name;
+
+  // xyz and t are likely useful here
+  for (auto & v : _xyzt)
+    variables += (variables.empty() ? "" : ",") + v;
+
+  // base function object
+  _func_F = std::make_shared<SymFunction>();
+
+  // set FParser internal feature flags
+  setParserFeatureFlags(_func_F);
+
+  // parse function
+  if (_func_F->Parse(_expression, variables) >= 0)
+    mooseError("Invalid function\n", _expression, "\nError:\n", _func_F->ErrorMsg());
+
+  // optimize
+  if (!_disable_fpoptimizer)
+    _func_F->Optimize();
+
+  // just-in-time compile
+  if (_enable_jit)
+  {
+    // let rank 0 do the JIT compilation first
+    if (_communicator.rank() != 0)
+      _communicator.barrier();
+
+    _func_F->JITCompile();
+
+    // wait for ranks > 0 to catch up
+    if (_communicator.rank() == 0)
+      _communicator.barrier();
+  }
+
+  // reserve storage for parameter passing buffer
+  _func_params.resize(_n_functors + 4);
+
+  // keep pointers to the functors
+  for (const auto & name : _functor_names)
+    _functors.push_back(&getFunctor<Real>(name));
```

Comment:
This kind of logic is repeated in a fair number of places
```
(moose-python) lindad@rod:/data/lindad/projects/moose/framework(avoid-refine-in-reinit-noop-30725)$ git grep --name-only JITCompile
include/utils/ADFParser.h
src/auxkernels/ArrayParsedAux.C
src/auxkernels/ParsedAux.C
src/auxkernels/ParsedVectorAux.C
src/materials/DerivativeParsedMaterialHelper.C
src/postprocessors/ParsedPostprocessor.C
src/scalarkernels/ParsedODEKernel.C
src/utils/ADFParser.C
src/utils/FunctionParserUtils.C
```
Can we make a helper function somewhere to help reduce the number of times we repeat this pattern? There is certainly some variation throughout these grep match occurrences but maybe the helper function could take arguments that support those variations